### PR TITLE
feat: avatar decorations

### PR DESCRIPTION
**Pull request by Snazzah**
Ref: ~~https://github.com/discord/discord-api-docs/pull/5723~~ https://github.com/discord/discord-api-docs/pull/6464

~~This also is based on new seasonal avatar decorations being prefixed with `v2_` to denote using an avatar decoration preset.~~

### CODE REVIEW

File: lib/gateway/Shard.js
Comment (TTtie) [inline lib/gateway/Shard.js]: The `user` variable is a User object:

```suggestion
                            avatarDecoration: user.avatarDecoration
```

File: lib/gateway/Shard.js
Comment (TTtie) [inline lib/gateway/Shard.js]: ditto

```suggestion
                        * @prop {String?} oldUser.avatarDecoration The hash of the user's avatar decoration, or null if no avatar decoration
```

**Review by TTtie — DISMISSED** [review]
Two small nits + please update the TS definitions (if you aren't able to, let me know and I'll update them for you), otherwise it's good to go once the appropriate docs PR is merged

File: lib/gateway/Shard.js
Comment (Snazzah) [reply lib/gateway/Shard.js]: The `PartialUser` type defines `avatar_decoration` already, wouldn't it be something like this?

```suggestion
                            avatar_decoration: user.avatarDecoration
```

Comment (TTtie) [reply lib/gateway/Shard.js]: Having taken a look at the `PartialUser` type, a majority of its usage seems to be with raw API data, leaving the `userUpdate` definition as being the odd one out. FWIW it'd be more worth it to use a separate `OldUser` type for this to keep things consistent with the rest of the library (with `avatarDecoration` camel-cased)

**Comment by TTtie** [comment]
Hey there! :wave:

If this PR is still alive, then it might be worth refactoring it to the latest iteration: https://github.com/discord/discord-api-docs/pull/6464

That one is going to stay for a while, I think. Whereas the ones seen before seemed to be seasonal testing rounds…

**Comment by Snazzah** [comment]
Will refactor with the latest iteration later this weekend.

**Comment by Snazzah** [comment]
Sidenote: I removed `dynamicAvatarDecorationURL` from the User class since it seems that avatar decoration presets don't use size query strings

File: lib/gateway/Shard.js
Comment (TTtie) [inline lib/gateway/Shard.js]: Probably forgot to update that here

```suggestion
                    if(user && (user.username !== packet.d.user.username
                        || user.discriminator !== packet.d.user.discriminator
                        || user.avatar !== packet.d.user.avatar
                        || user.globalName !== packet.d.user.global_name
                        || user.avatarDecorationData?.asset !== packet.d.user.avatar_decoration_data?.asset
                    )) {
```

File: lib/gateway/Shard.js
Comment (TTtie) [inline lib/gateway/Shard.js]: `avatarDecoration` -> `avatarDecorationData`
```suggestion
                            avatarDecorationData: user.avatarDecorationData
```

File: lib/gateway/Shard.js
Comment (TTtie) [inline lib/gateway/Shard.js]: `avatarDecoration` -> `avatarDecorationData`
```suggestion
                        * @prop {Object?} oldUser.avatarDecorationData The data of the user's avatar decoration, or null if there is no avatar decoration
```

File: lib/gateway/Shard.js
Comment (TTtie) [inline lib/gateway/Shard.js]: Use camel-case to be consistent with the rest of the structure
```suggestion
                            avatarDecorationData: user.avatarDecorationData,
```

File: lib/structures/User.js
Comment (TTtie) [inline lib/structures/User.js]: `@prop` declarations are not to be used – they were removed in #113 (the suggestion below reverts the change)

```suggestion
* @extends Base
```

File: lib/structures/User.js
Comment (TTtie) [inline lib/structures/User.js]: It would be probably worth to declare the field as nullable, possibly even its own typedef at the end of the file?
```suggestion
             * @type {Object?}
```

File: lib/structures/User.js
Comment (TTtie) [inline lib/structures/User.js]: Ditto
```suggestion
     * @type {String?}
```

File: lib/structures/Member.js
Comment (TTtie) [inline lib/structures/Member.js]: Ditto
```suggestion
     * @type {String?}
```

**Review by TTtie — DISMISSED** [review]
A few nitpicks, otherwise, LGTM

**Review by TTtie — APPROVED** [review]
LGTM. BTW, I rebased the branch on top of latest dev, please keep that in mind if you'd like to change something more.

**Pull request merged**
